### PR TITLE
Prune EvalModelTemplate (3/n)

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of #4501

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: tests/trainer/test_dataloaders.py [hunk 1]
```diff
@@ -552,103 +527,36 @@ def test_dataloaders_with_fast_dev_run(tmpdir, fast_dev_run):
...
-    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, val_check_interval=0.5)
-
-    with pytest.raises(MisconfigurationException, match="using an IterableDataset"):
-        trainer.fit(model)
+    trainer.fit(model, val_dataloaders=eval_dataloader)
+    assert len(trainer.val_dataloaders) == 1
+    assert trainer.val_dataloaders[0] == eval_dataloader
 
-
-def test_val_inf_dataloader_error(tmpdir):
```
Comment (rohitgr7) [inline tests/trainer/test_dataloaders.py:587]: these deleted tests are not parametrized below.

File: tests/trainer/test_dataloaders.py [hunk 2]
```diff
@@ -1020,83 +923,74 @@ def train_dataloader(self):
...
+        model, f"{stage.dataloader_prefix}_dataloader", lambda: dataloader_wrapper(DataLoader(RandomDataset(32, 64)))
+    )
+    trainer_kwargs = {"default_root_dir": tmpdir, "max_epochs": 1, f"limit_{stage.dataloader_prefix}_batches": 0.5}
+    trainer = Trainer(**trainer_kwargs)
+    trainer_fn = "fit" if stage == RunningStage.TRAINING else stage.value
 
-    with pytest.raises(MisconfigurationException, match="using an IterableDataset"):
+    with pytest.raises(MisconfigurationException, match=r"using an IterableDataset .* must be `0.0`, `1.0`"):
+        getattr(trainer, trainer_fn)(model)
         trainer.fit(model)
```
Comment (awaelchli) [inline tests/trainer/test_dataloaders.py:991]: there are two trainer calls here, but the raised error can only occur in one of them.

**Review by awaelchli — APPROVED** [review]

File: tests/trainer/test_dataloaders.py (hunk 2 above)
Comment (rohitgr7) [reply tests/trainer/test_dataloaders.py]: good catch! forgot to remove it

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]
